pandas/tests/indexes/multi/test_constructor.py

```diff
@@ -472,3 +472,83 @@ def test_from_tuples_with_tuple_label():
     idx = pd.MultiIndex.from_tuples([(2, 1), (4, (1, 2))], names=('a', 'b'))
     result = pd.DataFrame([2, 3], columns=['c'], index=idx)
     tm.assert_frame_equal(expected, result)
+
+
+def test_from_frame():
+    df = pd.DataFrame([['a', 'a'], ['a', 'b'], ['b', 'a'], ['b', 'b']],
+                      columns=['L1', 'L2'])
+    expected = pd.MultiIndex.from_tuples([('a', 'a'), ('a', 'b'),
+                                          ('b', 'a'), ('b', 'b')],
+                                         names=['L1', 'L2'])
+    result = pd.MultiIndex.from_frame(df)
+    tm.assert_index_equal(expected, result)
+
+
+@pytest.mark.parametrize('squeeze,input_type,expected', [
+    (True, 'multi', pd.MultiIndex.from_tuples([('a', 'a'), ('a', 'b'),
+                                               ('b', 'a'), ('b', 'b')],
+                                              names=['L1', 'L2'])),
+    (True, 'single', pd.Index(['a', 'a', 'b', 'b'], name='L1')),
+    (False, 'multi', pd.MultiIndex.from_tuples([('a', 'a'), ('a', 'b'),
+                                                ('b', 'a'), ('b', 'b')],
+                                               names=['L1', 'L2'])),
+    (False, 'single', pd.MultiIndex.from_tuples([('a',), ('a',),
+                                                 ('b',), ('b',)],
+                                                names=['L1']))
+])
+def test_from_frame_squeeze(squeeze, input_type, expected):
+    if input_type == 'multi':
+        df = pd.DataFrame([['a', 'a'], ['a', 'b'], ['b', 'a'], ['b', 'b']],
+                          columns=['L1', 'L2'])
+    elif input_type == 'single':
+        df = pd.DataFrame([['a'], ['a'], ['b'], ['b']], columns=['L1'])
+
+    result = pd.MultiIndex.from_frame(df, squeeze=squeeze)
+    tm.assert_index_equal(expected, result)
+
+
+def test_from_frame_non_frame():
```

Comment:
can you parametrize over some common input types (2-d np array, Index, Series, scalar)